canonical-web-and-design/maas-ui
# feat(ui): add test traffic lights to machine details tabs

## Done

- Renamed and removed some components that all did the same thing - showing an icon that represents a test's status with some text
  - `ScriptStatus` is now simply an icon and optionally some text to show beside it. It handles both `ScriptResultStatus` and `TestStatusStatus` types separately because as far as I can tell there's no way to have a "union" enum in TS.
  - `MachineTestStatus` is specifically for the machine list, which pretty much just wraps `ScriptStatus` with a tooltip if the tests failed and does not show an icon for passed tests.
- Added test traffic lights to the machine details tab using the new `ScriptStatus` component.
- Converted a bunch of machine list components to TS

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the react machine details page for a machine and check that the icons next to "Commissioning" and "Tests" match the ones in the angular machine details page
- Check that the icons are still correct for the commissioning/tests tabs and that failed tests show up in the machine list with a tooltip.

## Fixes

Fixes #2236
